test/core/end2end/fixtures/http_proxy_fixture.c

```diff
@@ -304,6 +307,45 @@ static void on_server_connect_done(grpc_exec_ctx* exec_ctx, void* arg,
                       &conn->on_write_response_done);
 }
 
+/**
+ * Parses the proxy auth header value to check if it matches :-
+ * Basic <base64_encoded_user_cred>
+ * Returns true if it matches, false otherwise
+ */
+static bool proxy_auth_header_matches(grpc_exec_ctx *exec_ctx,
+                                      char *proxy_auth_header_val) {
+  if(proxy_auth_header_val == NULL) {
+    return false;
+  }
+  char **auth_header_strs;
+  size_t auth_header_nstrs;
+  bool auth_header_matches = false;
+  // Split the auth header value on space
+  gpr_string_split(proxy_auth_header_val, " ", &auth_header_strs,
+                   &auth_header_nstrs);
+  if(auth_header_nstrs != 2) {
+    goto done;
+  }
+  // Authentication type should be Basic
+  if(strcmp(auth_header_strs[0], "Basic") != 0) {
+    goto done;
+  }
+  // should match GRPC_END2END_HTTP_PROXY_TEST_CONNECT_CRED after decoding
+  grpc_slice decoded_slice =
+      grpc_base64_decode(exec_ctx, auth_header_strs[1], 0);
+  if(grpc_slice_str_cmp(
+      decoded_slice, GRPC_END2END_HTTP_PROXY_TEST_CONNECT_CRED) != 0) {
```

Comment:
As mentioned elsewhere, we should use the value of the channel arg instead of hard-coding this value.